payjoin/src/uri.rs

```diff
@@ -24,7 +24,7 @@ impl MaybePayjoinExtras {
 
 #[derive(Clone)]
 pub struct PayjoinExtras {
-    pub(crate) endpoint: Url,
+    pub endpoint: Url,
```

Comment:
this is useful in testing where we want to mutate the pjuri in order to change test direction(ie start with a pj uri and then maybe change it to something different) example: https://github.com/lightningdevkit/ldk-node/pull/257/files#diff-ebba69f49eedcd3dd6a656eff964c6184b6d29c02fbad6a49d3b41c87cc3de79R192